browse/services/listing/__init__.py

```diff
@@ -1,32 +1,268 @@
 """Serves lists of articles for categories and time periods."""
 
-from typing import cast, Any
-from browse.services.listing.base_listing import ListingService
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from datetime import date, datetime, timedelta
+from time import mktime
+from typing import Any, List, Literal, Optional, Tuple, Union, cast
+from wsgiref.handlers import format_date_time
 
 
-def get_listing_service() -> ListingService:
+def get_listing_service() -> "ListingService":
     """Get the listing service configured for the app context."""
     from browse.config import settings
     from flask import g
-    if 'listing_service' not in g:
-        g.listing_service = settings.DOCUMENT_LISTING_SERVICE(settings, g)  # pylint disable:E1102
+
+    if "listing_service" not in g:
+        g.listing_service = settings.DOCUMENT_LISTING_SERVICE(
+            settings, g
+        )  # pylint disable:E1102
 
     return cast(ListingService, g.listing_service)
 
 
-def fs_listing(settings: Any, _: Any) -> ListingService:
+def fs_listing(settings: Any, _: Any) -> "ListingService":
     """Factory function for filesystem-based listing service."""
     from .fs_listings import FsListingFilesService
+
     return FsListingFilesService(settings.DOCUMENT_LISTING_PATH)
 
-def db_listing(settings: Any, _: Any) -> ListingService:
+
+def db_listing(settings: Any, _: Any) -> "ListingService":
     """Factory function for DB backed listing service."""
-    from .db_listing_impl import DBListingService
     from browse.services.database import models
-    #maybe pass in the specific classes for the tables we need?
+
+    from .db_listing_impl import DBListingService
+
+    # maybe pass in the specific classes for the tables we need?
     return DBListingService(models.db)
 
-def fake(settings: Any, _: Any) -> ListingService:
+
+def fake(settings: Any, _: Any) -> "ListingService":
     """Factory function for fake listing service."""
     from .fake_listings import FakeListingFilesService
+
     return FakeListingFilesService()
+
+
+ListingTypes = Literal["new", "cross", "rep"]
+
```

Comment:
Below are the new listing classes that are class based versions of the previous TypedDict types. Classes with the attribute syntax are more pleasant to work with.